server/service/queries_test.go

```diff
@@ -114,6 +114,239 @@ func TestListQueries(t *testing.T) {
 	}
 }
 
+func TestQueryPayloadValidationCreate(t *testing.T) {
+	ds := new(mock.Store)
+	ds.NewQueryFunc = func(ctx context.Context, query *fleet.Query, opts ...fleet.OptionalArg) (*fleet.Query, error) {
+		return query, nil
+	}
+	ds.NewActivityFunc = func(ctx context.Context, user *fleet.User, activity fleet.ActivityDetails) error {
+		act, ok := activity.(fleet.ActivityTypeCreatedSavedQuery)
+		assert.True(t, ok)
+		assert.NotEmpty(t, act.Name)
+		return nil
+	}
+	svc, ctx := newTestService(t, ds, nil, nil)
+
+	testCases := []struct {
+		name         string
+		queryPayload fleet.QueryPayload
+		shouldErr    bool
+	}{
+		{
+			"All valid",
+			fleet.QueryPayload{
+				Name:     ptr.String("test query"),
+				Query:    ptr.String("select 1"),
+				Logging:  ptr.String("snapshot"),
+				Platform: ptr.String(""),
+			},
+			false,
+		},
+		{
+			"Invalid  - empty string name",
+			fleet.QueryPayload{
+				Name:     ptr.String(""),
+				Query:    ptr.String("select 1"),
+				Logging:  ptr.String("snapshot"),
+				Platform: ptr.String(""),
+			},
+			true,
+		},
+		{
+			"Empty SQL",
+			fleet.QueryPayload{
+				Name:     ptr.String("bad sql"),
+				Query:    ptr.String(""),
+				Logging:  ptr.String("snapshot"),
+				Platform: ptr.String(""),
+			},
+			true,
+		},
+		{
+			"Invalid logging",
+			fleet.QueryPayload{
+				Name:     ptr.String("bad logging"),
+				Query:    ptr.String("select 1"),
+				Logging:  ptr.String("hopscotch"),
+				Platform: ptr.String(""),
+			},
+			true,
+		},
+		{
+			"Unsupported platform",
+			fleet.QueryPayload{
+				Name:     ptr.String("invalid platform"),
+				Query:    ptr.String("select 1"),
+				Logging:  ptr.String("differential"),
+				Platform: ptr.String("charles"),
+			},
+			true,
+		},
+		{
+			"Missing comma",
+			fleet.QueryPayload{
+				Name:     ptr.String("invalid platform"),
+				Query:    ptr.String("select 1"),
+				Logging:  ptr.String("differential"),
+				Platform: ptr.String("darwin windows"),
+			},
+			true,
+		},
+		{
+			"Unsupported platform 2",
```

Comment:
nit: 
```suggestion
			"Unsupported platform sphinx",
```

for a second i thought there's a valid linux distro sphinx